### ISSUE TITLE
Page::inc_ref_count violates the reference counting invariant.

### ISSUE THREAD

**Issue by rikosellic** [asterinas/asterinas#1260]
### Describe the bug

During formally verifying the page system I find that function` inc_ref_count ` violates the invariant that the `ref_count` must be greater than or equal to the number of page owners.
```
pub(in crate::mm) unsafe fn inc_ref_count(paddr: Paddr) {
        let page = unsafe { ManuallyDrop::new(Self::from_raw(paddr)) };
        let _page = page.clone();
    }
```
After the first step, the page owner increases by one but the reference count does not change. While the precondition of this function states that the caller must already hold a reference count so that the invariant still holds at the end of the function, it is dubious and unnecessary to break the invariant here, especially considering concurrent operations.

### Expected behavior
I suggest that a simple `self.ref_count().fetch_add(1, Ordering::Relaxed);` is enough, which can do the same thing more intuitively. It will not break the invariant, and formal verification of the function will work smoothly.
<!-- A cle

**Comment by junyang-zh** [comment]
The `self.ref_count().fetch_add(1, Ordering::Relaxed);` approach is indeed better than the current implementation in the sense of maintainability. No damaging effects are yet confirmed. Perhaps you can make a PR fixing it?

**Comment by rikosellic** [comment]
> The `self.ref_count().fetch_add(1, Ordering::Relaxed);` approach is indeed better than the current implementation in the sense of maintainability. No damaging effects are yet confirmed. Perhaps you can make a PR fixing it?

I realized that the original design is to allow increasing ref_count at an arbitrary physical address, the design of the function may require reconsideration.

**Comment by rikosellic** [comment]
Maybe something like 
```
pub(in crate::mm) unsafe fn inc_ref_count(paddr: Paddr) {
        let ref_count = unsafe {&(*(paddr as *const MetaSlot)).ref_count};
        ref_count.fetch_add(1, Ordering::Relaxed);
    }
```
I'm still testing and verifying this version.

**Comment by rikosellic** [comment]
Fixed in PR #1269.

**Comment by rikosellic** [comment]
Fixed in #1271
